processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -341,7 +435,18 @@ public void advance()
                       if (Thread.interrupted()) {
                         throw new QueryInterruptedException(new InterruptedException());
                       }
-                      cursorOffset.increment();
+                      cursorOffsetHolder.get().increment();
+
+                      while (!isDone()) {
```

Comment:
@jon-wei cool, good to know. Could you also try one reading from a long column instead of a string column (like what a timeseries with a single sum would do)? Overhead for reading a string is higher than a long, so that might make it harder to notice overheads in the cursor itself.